staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -414,13 +400,22 @@ func (qs *queueSet) chooseQueueIndexLocked(hashValue uint64, descr1, descr2 inte
 	bestQueueIdx := -1
 	bestQueueLen := int(math.MaxInt32)
 	// the dealer uses the current desired number of queues, which is no larger than the number in `qs.queues`.
-	qs.dealer.Deal(hashValue, func(queueIdx int) {
-		thisLen := len(qs.queues[queueIdx].requests)
-		klog.V(7).Infof("QS(%s): For request %#+v %#+v considering queue %d of length %d", qs.config.Name, descr1, descr2, queueIdx, thisLen)
-		if thisLen < bestQueueLen {
-			bestQueueIdx, bestQueueLen = queueIdx, thisLen
-		}
-	})
+	if qs.config.Dealer != nil {
+		qs.config.Dealer.Deal(hashValue, func(queueIdx int) {
+			if queueIdx < 0 || queueIdx >= len(qs.queues) {
+				return
+			}
+			thisLen := len(qs.queues[queueIdx].requests)
+			klog.V(7).Infof("QS(%s): For request %#+v %#+v considering queue %d of length %d", qs.config.Name, descr1, descr2, queueIdx, thisLen)
+			if thisLen < bestQueueLen {
+				bestQueueIdx, bestQueueLen = queueIdx, thisLen
+			}
+		})
+	}
+	if bestQueueIdx < 0 {
+		klog.V(6).Infof("QS(%s): dealer dealt no valid cards", qs.config.Name)
```

Comment:
I'd honestly rather crash, since this is config that has no reason to ever be wrong and the system will behave rather differently than it should if we do this.

Reply:
done